docs/zts_access_token_guide.md

```diff
@@ -0,0 +1,380 @@
+
+An ID token is a [JSON web token (JWT)](https://tools.ietf.org/html/rfc7519)
+encoded in Base64 URL-encoded format that contains a header, payload, and
+signature. The ID token, if enabled, can be issued for clients to authenticate
+themselves along with their access token to the requested Athenz service.
+ZTS Server never issues ID Tokens on its own. They're always returned along
+with access tokens.
+
+### ID Token Header
+
+The ID token header contains the Athenz ZTS Server Private key id and the
+algorithm used to sign the payload.
+
+```
+{
+  "alg": "ES256",
+  "kid": "<zts server private key id>"
+}
+```
+
+### ID Token Payload
+
+The ID Token Payload contains the following claims:
+
+```
+ver : version of the id token
+iss : token issuer
+aud : the audience (the Athenz Service name) that the id token is intended for
+sub : subject of the id token - athenz pricnipal name requesting the token
+iat : token issue time in seconds (Unix time)
+exp : token expiry time in seconds (Unix time)
+auth_time: authentication time in seconds (Unix time)
+```
+
+Here is an example of an id token that `alpha.api` retrieved for its
+authorization request to `beta.backend` service.
+
+```
+{
+  "ver": 1,
+  "iss": "athenz",
+  "aud": "beta.backend",
+  "sub": "alpha.api",
+  "iat": 1554491974,
+  "auth_time": 1554491974,
+  "exp": 1554495574
+}
+```
+
+### ID Token Validation
+
+You can use any standards based JWT library to validate the id token
+generated by ZTS Server. You can also implement your own verifier by
+following the [RFC7519: Section 7.2:  Validating a JWT](https://tools.ietf.org/html/rfc7519).
+
+## Fetching Access Tokens with Java Client Library
+
+First you need to update your Java project `pom.xml` file to indicate
+the dependency on the Athenz zts java client library. Checkout the
+[Bintray ZTS Java Client Package](https://bintray.com/yahoo/maven/athenz-zts-java-client/)
+page to make sure you're using the latest release version (You must use
+1.8.37 or newer version of the client library). 
+
+### ZTS Client Object
+
+ZTS Client Library provides several constructors. The recommended
+approach is to use the an SSL context that includes service's Athenz
+issued CA certificate.
+
+ZTSClient object must be closed to release any allocated resources.
+ZTSClient class implements Closeable interface. However, to correctly
+handle auto-refresh of access tokens, the client used to fetch
+the token cannot be closed since the background tasks need to use
+the same client to refresh the access token. Our general recommendation
+is that you create a single ZTSClient object and use that for all
+requests (it is thread safe) and then close the client when your
+application is shutting down.
+
+** Important **
+
+During the shutdown of the application, `ZTSClient.cancelPrefetch()`
+must be called to stop the timer thread that automatically fetches
+and refreshes any cached tokens in the ZTS Client.
+
+#### Athenz Issued CA certificate
```

Comment:
Let's reword this section to remove the internal assumption of identity being available out-of-the-box.